### ISSUE TITLE
Create new instance of STI model

### ISSUE THREAD

**Issue by rctneil** [avo-hq/avo#740]
Hi Adrian,

As, you know I have a `Person` model and a `Spouse` model.  `Spouse` inherits from `Person`.

When I go to create a new person, how can I specify what type this new person should be?

I tried adding a select like below, but I just get an error saying "The single-table inheritance mechanism failed to locate the subclass: 'person'. This error is raised because the column 'type' is reserved for storing the class in case of inheritance. Please rename this column if you didn't intend it to be used for storing the inheritance class or overwrite Person.inheritance_column to use another column for that information."

```
  field :type,
        as: :select,
        name: 'Type',
        only_on: [:forms],
        options: ->(model: m, resource: r, view: v, field: f) {
            [['Person', 'person'], ['Spouse', 'spouse']]
        }
```

On a similar line,  if there is some form of dropdown to be able to select the `type` of Person being created, can I hide and show fields further down the create form (because some fields only relate to a spouse and not when it's a basic person etc.

Many thanks,
Neil

**Comment by adrianthedev** [comment]
Hey @rctneil,

So that error (`The single-table inheritance mechanism ...`) comes from Rails and not Avo itself. Not sure what triggered that.
Definitely, you should fix that first, before attempting to make Avo work. It's an underlying issue with the app. Maybe try adding the `foreign_key` to your associations like [here](https://github.com/avo-hq/avodemo/blob/main/app/models/person.rb#L3) and [here](https://github.com/avo-hq/avodemo/blob/main/app/models/person.rb#L3). 

Also, I don't think that you need to specify the `Person` model with `Person` type. You leave that blank and it's automatically the type of `Person`. Only new types must be declared (eg: `Spouse`). I posted a screenshot of my database records. Those with `type` empty are `Person`.

TBH, the type selector you have here looks pretty good. Our only tweak is that we passed the types directly with the first letter uppercased `field :type, as: :select, name: "Type", options: { Spouse: "Spouse" }` (only `Spouse` as type). 
Is that on the `Person` model? Please check out our demo app source code. You can even clone it and run it locally to try out more scenarios.
I posted a few links below to how we structured it with the demo app. I modeled the app to your specifications, so it should work for you too. Maybe you can have a look and see if there are differences between the demo app and yours. 


Demo link: https://avodemo.herokuapp.com/avo/resources/people/new
Code link: https://github.com/avo-hq/avodemo/blob/main/app/avo/resources/person_resource.rb#L11

![image](https://user-images.githubusercontent.com/1334409/159878343-aa72f7a1-4912-4e66-b4eb-9d0bb29fca96.png)

**Comment by adrianthedev** [comment]
Does this work @rctneil?

**Comment by rctneil** [comment]
Hi @adrianthedev   This was 100% my own stupid fault!    Basically, when testing out creating a new Person, because I had the "Type" field in my resource with options for both "person" and "spouse", I selected "person" and it did create a record, but it set the type to "person", then as soon as I tried to view the Index page for People, it failed trying to find a subclass of "person" (quite rightly).   I didn't notice it had created that record and wrongly blamed Avo!

Removing that rogue record and ensuring the "Type" field is only changed to Spouse (when creating a Spouse) solved the issue!

Thanks again!
